### ISSUE TITLE
early stopping occurs too early

### ISSUE THREAD

**Issue by hermancollin** [ivadomed/ivadomed#867]
### Current behavior

Early stopping occurs earlier than expected. Current implementation:
```python
# EARLY STOPPING
if epoch > 1:
    val_diff = (val_loss_total_avg_old - val_loss_total_avg) * 100 / abs(val_loss_total_avg)
    if val_diff < training_params["training_time"]["early_stopping_epsilon"]:
        patience_count += 1
    if patience_count >= training_params["training_time"]["early_stopping_patience"]:
        logger.info("Stopping training due to {} epochs without improvements".format(patience_count))
        break
```
As we can see, the patience_count never resets.

### Expected behavior
 

[The doc](https://ivadomed.org/en/latest/configuration_file.html#training-time) states that early stopping should occur when the validation loss difference is inferior to the `early_stopping_epsilon` criteria for `early_stopping_patience` **consecutive** epochs. 

However, because the `patience_count` variable is never resetted, early stopping is triggered whenever there weren't any improvement for `early_stopping_patience` epochs (probably not consecutive epochs). 

Also, maybe the doc should mention that relative (not absolute) difference is implemented for early stopping, as we can see above. 

### Steps to reproduce

Train any model with early stopping parameters. For me, the parameters used were:
```json
"training_time": {
            "num_epochs": 200,
            "early_stopping_patience": 50,
            "early_stopping_epsilon": 0.001
}
```

### Installed packages

<details close>
<summary> Output of <code>pip freeze</code> </summary>

```
absl-py==0.13.0
aiocontextvars==0.2.2
bids-validator==1.7.3
cachetools==4.2.2
certifi==2021.5.30
charset-normalizer==2.0.2
click==8.0.1
contextvars==2.4
csv-diff==1.1
cycler==0.10.0
dataclasses==0.8
decorator==4.4.2
dictdiffer==0.8.1
docopt==0.6.2
future==0.18.2
google-auth==1.33.0
google-auth-oauthlib==0.4.4
grpcio==1.38.1
h5py==2.10.0
idna==3.2
imageio==2.9.0
immutables==0.15
importlib-metadata==4.6.1
-e git+https://github.com/ivadomed/ivadomed.git@7a595ae39e94085837582f7176b8ec59bd9a2bc9#egg=ivadomed
joblib==1.0.1
kiwisolver==1.3.1
loguru==0.5.3
Markdown==3.3.4
matplotlib==3.3.4
networkx==2.5.1
nibabel==3.2.1
num2words==0.5.10
numpy==1.19.5
oauthlib==3.1.1
onnxruntime==1.4.0
packaging==21.0
pandas==1.1.5
patsy==0.5.1
Pillow==8.3.1
pkg-resources==0.0.0
protobuf==3.17.3
pyasn1==0.4.8
pyasn1-modules==0.2.8
pybids==0.13.1
pyparsing==2.4.7
python-dateutil==2.8.2
pytz==2021.1
PyWavelets==1.1.1
requests==2.26.0
requests-oauthlib==1.3.0
rsa==4.7.2
scikit-image==0.17.2
scikit-learn==0.24.2
scipy==1.5.4
seaborn==0.11.1
six==1.16.0
SQLAlchemy==1.3.24
tensorboard==2.5.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
threadpoolctl==2.2.0
tifffile==2020.9.3
torch==1.5.0+cu92
torchvision==0.6.0+cu92
tqdm==4.61.2
typing-extensions==3.10.0.0
urllib3==1.26.6
Werkzeug==2.0.1
zipp==3.5.0
```

</details>

**Comment by charleygros** [comment]
@hermancollin : Very good points, thanks for reporting!

Feel free to implement the proposed changes via a Pull Request if you have some time, that would be really appreciated!
Otherwise, no worries, simply let us know and we will make sure this bug get fixed very soon.

Cheers.
